segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -167,6 +175,57 @@ public void setupAppend(SetupAppend setupAppend) {
                 });
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> setupTokenExpiryTask(SetupAppend setupAppend) {
+        String segment = setupAppend.getSegment();
+        UUID writerId = setupAppend.getWriterId();
+        long requestId = setupAppend.getRequestId();
+
+        Duration expiryDuration = this.durationToExpiry(setupAppend);
+
+        if (expiryDuration == null) {
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.delayedTask(() -> {
+                if (isSetupAppendCompleted(segment, writerId)) {
+                    try {
+                        log.debug("Informing writer {} that sent request {}, about token expiry for segment {}",
+                                writerId, requestId, segment);
+                        connection.send(new WireCommands.AuthTokenCheckFailed(setupAppend.getRequestId(),
```

Comment:
Can we avoid sending this if the connection is closed? A connection closing should ensure all the pending tasks are cancelled.

Reply:
The `ServerConnection` interface doesn't seem to have a way to check if the connection is closed. My understanding is that if the connection is closed, sending a message over it will result in an exception being thrown, which we are catching and ignoring here. 

What do you say? Am I missing something else?

Reply:
The connection can be closed externally or when due to a connection.close() in by the appendProcessor causing these delayed tasks to be unnecessarily executed.

Reply:
I've now introduced an `isClosed()` method to the connection, so that we the task checks for it before trying to send the token expiry notification. This makes it a bit more efficient. However, the core issue remains: whether/when/how do we cancel the task?